pkg/url/types.go

```diff
@@ -48,3 +55,154 @@ const (
 	// StateTypeLocallyDeleted means that URL was deleted from the local config, but it is still present on the cluster/container
 	StateTypeLocallyDeleted = "Locally Deleted"
 )
+
+func NewURLFromKubernetesIngress(ki *unions.KubernetesIngress) URL {
+	if ki.IsGenerated() {
+		return URL{}
+	}
+	u := URL{
+		TypeMeta: metav1.TypeMeta{Kind: "url", APIVersion: apiVersion},
+	}
+	if ki.NetworkingV1Ingress != nil {
+		u.ObjectMeta = metav1.ObjectMeta{Name: ki.NetworkingV1Ingress.Labels[urlLabels.URLLabel]}
+		u.Spec = URLSpec{
+			Host:   ki.NetworkingV1Ingress.Spec.Rules[0].Host,
+			Port:   int(ki.NetworkingV1Ingress.Spec.Rules[0].HTTP.Paths[0].Backend.Service.Port.Number),
+			Secure: ki.NetworkingV1Ingress.Spec.TLS != nil,
+			Path:   ki.NetworkingV1Ingress.Spec.Rules[0].HTTP.Paths[0].Path,
+			Kind:   localConfigProvider.INGRESS,
+		}
+		if len(ki.NetworkingV1Ingress.Spec.TLS) > 0 {
+			u.Spec.TLSSecret = ki.NetworkingV1Ingress.Spec.TLS[0].SecretName
+		}
+		if u.Spec.Secure {
+			u.Spec.Protocol = "https"
+		} else {
+			u.Spec.Protocol = "http"
+		}
+	} else if ki.ExtensionV1Beta1Ingress != nil {
+		u.ObjectMeta = metav1.ObjectMeta{Name: ki.ExtensionV1Beta1Ingress.Labels[urlLabels.URLLabel]}
+		u.Spec = URLSpec{
+			Host:   ki.ExtensionV1Beta1Ingress.Spec.Rules[0].Host,
+			Port:   int(ki.ExtensionV1Beta1Ingress.Spec.Rules[0].HTTP.Paths[0].Backend.ServicePort.IntVal),
+			Secure: ki.ExtensionV1Beta1Ingress.Spec.TLS != nil,
+			Path:   ki.ExtensionV1Beta1Ingress.Spec.Rules[0].HTTP.Paths[0].Path,
+			Kind:   localConfigProvider.INGRESS,
+		}
+		if len(ki.ExtensionV1Beta1Ingress.Spec.TLS) > 0 {
+			u.Spec.TLSSecret = ki.ExtensionV1Beta1Ingress.Spec.TLS[0].SecretName
+		}
+		if u.Spec.Secure {
+			u.Spec.Protocol = "https"
+		} else {
+			u.Spec.Protocol = "http"
+		}
+	}
+	return u
+}
+
+// Get returns URL definition for given URL name
+func (urls URLList) Get(urlName string) URL {
+	for _, url := range urls.Items {
+		if url.Name == urlName {
+			return url
+		}
+	}
+	return URL{}
+
+}
+
+func (u URL) GetKubernetesIngress(serviceName string) *unions.KubernetesIngress {
```

Comment:
This function is unused and seems to initiate both types of ingresses which seems weird.

Reply:
I have seen places where url is converted to ingress. So I guess that is why i have prepared this function. The lower layers can decide which to use based on support

Reply:
Il update this further if needed during second pr

Reply:
But currently it's initiating both types of ingress which is weird. We should remove it for now and add it later if needed with a better solution.

Reply:
ok il move this function to my notes